### PR TITLE
Reimplement a synchronous method - Fixes #13

### LINKED ISSUE #13: sync methods still relevant
This goes back a ways, but the changes to the API moving to an async, promise-based pattern in #9 are causing some headache with regard to updating. specifically, webpack-dev-server utilizes this module, and while a refactor to be more promise-friendly is in the works, it'll be some time before the project can use 2.x of internal-ip without a synchronous method. I'd like to suggest bringing back a synchronous method, even if it's not as accurate and contained a disclaimer, to allow use of the newer versions in apps/modules/projects that may not be able to leverage the updated async API as yet.

It would seem that @silverwind 's `default-gateway` module, and it's use of `execa`'s async patterns is the sole reason that the API is 100% async. It would be reasonable to provide sync methods via `default-gateway` if that module would also leverage [`execa.sync`](https://github.com/sindresorhus/execa#execasyncfile-arguments-options)

Of course, setting the dependency to the hard 1.2.0 version is always a solution. But providing synchronous methods here as well would be a big win.

### PR DESCRIPTION
**Pull request by shellscape**
This PR makes use of a recent update to `default-gateway` which provides synchronous methods, to reimplement sync support in this module.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -8,29 +8,41 @@ const defaults = {
...
+function promise(family) {
+	return defaultGateway[family]().then(result => {
+		return match(result.gateway, family);
 	}).catch(() => defaults[family]);
 }
 
-module.exports.v6 = () => internalIp('v6');
-module.exports.v4 = () => internalIp('v4');
+function sync(family) {
+	const result = defaultGateway[family].sync();
```
Comment (silverwind) [inline index.js]: you need to wrap this line in a `try...catch` and return the defaults in case of error.

File: index.js [hunk 2]
```diff
@@ -8,29 +8,41 @@ const defaults = {
...
-				const prefix = ipaddr.parse(addr.netmask).prefixLengthFromSubnetMask();
-				const net = ipaddr.parseCIDR(`${addr.address}/${prefix}`);
-
-				if (net[0].kind() === gatewayIp.kind() && gatewayIp.match(net)) {
-					ret = net[0].toString();
-				}
-
-				return Boolean(ret);
-			});
+function match(gateway, family) {
```
Comment (silverwind) [inline index.js]: maybe `findIp` might be a better name.

File: index.js [hunk 3]
```diff
@@ -8,29 +8,41 @@ const defaults = {
...
+	const gatewayIp = ipaddr.parse(gateway);
+	let ret;
+
+	// Look for the matching interface in all local interfaces
+	Object.keys(interfaces).some(name => {
+		return interfaces[name].some(addr => {
+			const prefix = ipaddr.parse(addr.netmask).prefixLengthFromSubnetMask();
+			const net = ipaddr.parseCIDR(`${addr.address}/${prefix}`);
+
+			if (net[0].kind() === gatewayIp.kind() && gatewayIp.match(net)) {
```
Comment (silverwind) [inline index.js:22]: just noticed this: could you add a check for `net[0]` here, e.g. `if (net[0] && net[0].kind()`?

**Review by silverwind — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (shellscape) [reply index.js:22]: this is a direct copy paste from the code as it presently is in master. mentioning only to confirm that you want to piggy back on this PR for the change

File: index.js (hunk 2 above)
Comment (shellscape) [reply index.js]: from line 16 `// Look for the matching interface in all local interfaces` - that's why I chose the name. doesn't much matter to me what you think it should be, happy to change it, but I thought it was appropriate given that comment

File: index.js (hunk 1 above)
Comment (shellscape) [reply index.js]: will do

File: index.js (hunk 3 above)
Comment (silverwind) [reply index.js:22]: right, I can fix that later myself, no need to do it here.

File: index.js (hunk 2 above)
Comment (silverwind) [reply index.js]: Yeah, I still think `findIp` is more appropriate. The matching is only part of the process.

Comment (shellscape) [reply index.js]: aye, will change

**Review by silverwind — APPROVED** [review]

**Pull request merged**